### ISSUE TITLE
Suggest super functions marked with overload

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#868]
<h3> Environment data </h3>

-   Language Server version: 2021.1.2
-   OS and version: darwin x64
-   Python version: 3.9.1 
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

<h3> Expected behaviour </h3>

Pylance should suggest all functions of a super class which could be overwritten.

<h3> Actual behaviour </h3>

Only functions that don't have an `overload` decorator are suggested.

<h3> Logs </h3>
 --

<h3> Code Snippet / Additional information </h3>

`ArgumentParser.parse_args()`

```py
from argparse import ArgumentParser

class CustomArgParser(ArgumentParser):
    def parse_args
```

<img width="598" alt="Screen Shot 2021-01-22 at 23 21 59" src="https://user-images.githubusercontent.com/30130371/105555580-f9aebf80-5d09-11eb-9562-64ec0ce94cec.png">

[Typeshed - argparse.pyi](https://github.com/python/typeshed/blob/83c18117d5165056dc85fb90baf69c7ad846ebc0/stdlib/2and3/argparse.pyi#L170-L179)

Just to give another example: `bytes.__getitem__()`

```py
class CustomBytes(bytes):
    def get
```

<img width="571" alt="Screen Shot 2021-01-22 at 23 37 25" src="https://user-images.githubusercontent.com/30130371/105556705-d59fae00-5d0a-11eb-9aab-e3c3130bb3b5.png">

[Typeshed - builtins.pyi](https://github.com/python/typeshed/blob/83c18117d5165056dc85fb90baf69c7ad846ebc0/stdlib/3/builtins.pyi#L471-L474)

**Comment by erictraut** [comment]
Thanks for the suggestion. Currently, pylance offers to complete a function override for non-overloaded functions. 

Overloaded functions present an interesting problem, especially if the overloads come from a type stub file. The problem is that the type signature of the runtime function or method isn't specified in a type stub. So there's no way for pylance to know what signature to use in the completion.

If the base class is defined in a source file (rather than a stub), then it will have both overloads and a runtime implementation. In that case, Pylance could offer the runtime implementation's signature in the completion. But that wouldn't help in either of the cases you mentioned above because the type definitions for `ArgumentParser` and `bytes` come from type stubs.

**Comment by cdce8p** [comment]
I don't know if it would work, but would it be possible to add a function signature more than ones?

I theory I'm thinking about something like the hover hint below where you can cycle though the different overloads. However, I don't believe that's possible at the moment, that's why I'm suggesting multiple entries. To select the correct one a user could use the extended hint.

<img width="702" alt="Screen Shot 2021-01-23 at 00 06 36" src="https://user-images.githubusercontent.com/30130371/105558450-e2260580-5d0e-11eb-97a7-da40ab16e19c.png">

<img width="949" alt="Screen Shot 2021-01-23 at 00 09 38" src="https://user-images.githubusercontent.com/30130371/105558636-4f399b00-5d0f-11eb-97a3-179e0c91e15e.png">

**Comment by erictraut** [comment]
With overloaded functions, the type signature of the runtime implementation generally doesn't match any of the overload signatures. It's a combination of all of them.

**Comment by cdce8p** [comment]
I see. What about just suggesting the function arguments without type hints (or with Any) in those cases?
Those should be known from the overloads or am I getting something totally wrong here?

**Comment by erictraut** [comment]
I don't think that would work because overload signatures often omit some parameters (those that have default values in the runtime signature) or use *args/**kwargs.

**Comment by erictraut** [comment]
How often do you find that you need to override an overloaded method? I can't remember ever doing this myself, but maybe my usage differs from yours.

**Comment by cdce8p** [comment]
Not that often, so it certainly wouldn't be worth finding a solution just for me alone.

With all the issues I open it's mostly that, while working on some other project, I find these little things where I constantly imagine they should work but they don't. I know how hard it is to get good bug reports and if I open one maybe somebody finds it relevant enough and it will eventually get fixed. If it's something that really bothers me, I'm even willing to spend some of my free time to fix it myself if I can.

Just to get back to the topic, this one definitely is not :)
